clang/lib/Sema/SemaExpr.cpp

```diff
@@ -62,7 +62,34 @@
 
 using namespace clang;
 using namespace sema;
-
+class EnumNameVisitor : public RecursiveASTVisitor<EnumNameVisitor> {
+private:
+  std::string EnumName;
+public:
+  void VisitDeclRefExpr(DeclRefExpr *Node) {
+    const Decl *D = Node->getDecl();
+    // Check if the declaration referenced is an enum constant
+    if (const auto *ECD = dyn_cast<EnumConstantDecl>(D)) {
+      if (const auto *ED = dyn_cast<EnumDecl>(ECD->getDeclContext())) {
+        // Construct the fully qualified name
+        EnumName = ED->getName().str() + "::" + ECD->getName().str();
+        return; // No error for valid enum constants
```

Comment:
Why do you need this return? Even without this `return` statement, the function returns from here.